internal/maincmd/maincmd.go

```diff
@@ -1,3 +1,7 @@
+// Package maincmd implements a subset of the '$ rsync' CLI surface, namely that it can:
+// - serve as a server daemon over TCP or SSH (via SSH session stdin/stdout)
+// - act as "client" CLI for connecting to the server
+// - both "client" and "server" can act as the sender and the receiver (though not yet implemented).
```

Comment:
You asked:
> Maybe you could add a package-level comment that explains this while you’re at it?

I tried, let me know if this is inaccurate. I might still not understand the whole picture :smile:

Reply:
What you wrote is correct, but I suggest rephrasing the last point slightly for clarity. How about:

Not yet implemented: at a later point, both "client" and "server" will be able to act as rsync sender or receiver

Reply:
Good idea, I totally agree the comment was unclear